origins_workflow/src/Controller/ModerationStateController.php

```diff
@@ -113,6 +112,18 @@ private function transitionAllowed($entity, String $new_state) {
         $transition_allowed = TRUE;
       }
     }
+    elseif (($current_state == 'draft') && ($new_state == 'published')) {
+      // Is this user allowed to use the 'quick publish' transition ?
+      if ($current_user->hasPermission('use editorial transition quick_publish')) {
+        $transition_allowed = TRUE;
+      }
+    }
+    elseif (($current_state == 'published') && ($new_state == 'published')) {
+      // Is this user allowed to use the 'quick publish' transition ?
+      if ($current_user->hasPermission('use editorial transition quick_publish')) {
+        $transition_allowed = TRUE;
+      }
+    }
```

Comment:
Several other state transitions that weren't previously covered have been added.